src/controller.js

```diff
@@ -0,0 +1,312 @@
+				postalCode: inputs.postalCode,
+				country: inputs.country,
+			},
+			hasMultipleSites: false,
+			hasTestingFacilities: inputs.hasTestingFacilities,
+		};
+		console.log('submit_organization -> postData:', postData);
+		const req = {
+			url: `${API_HOST}/organizations/`,
+			data: JSON.stringify(postData),
+			type: 'POST',
+			success: responseData => {
+				console.log('submit_organization -> responseData:', responseData);
+				let category, subcategory;
+				if (inputs.type.length > 0) {
+					category = inputs.type.split('/')[0];
+					subcategory = inputs.type.split('/')[1];
+				}
+
+				console.log('submit_organization -> submit_site')
+				controller.submit_site({
+					modal_id: inputs.modal_id,
+					button_id: inputs.button_id,
+					org_id: responseData.id,
+					org_name: inputs.org_name,
+					isTesting: inputs.hasTestingFacilities,
+					email: inputs.email,
+					phone: inputs.phone,
+					contact_name: inputs.contact_name,
+					category,
+					subcategory,
+				});
+			},
+			error: e => {
+				console.log('Error from "submit_organization":', e);
+				console.log('Arguments from "submit_organization":', inputs);
+				$(inputs.button_id).removeClass('btn-loading');
+				$(inputs.modal_id).find('.alert-warning').removeClass('d-none');
+			},
+		};
+		jQuery.ajax(Object.assign(defaultRequestOpts, req));
+	},
+	submit_site: inputs => {
+		console.log('submit_site -> inputs:', inputs);
+		const postData = {
+			name: inputs.org_name,
+			isTesting: inputs.isTesting,
+			siteManagerContactInfo: {
+				phone: inputs.phone,
+				email: inputs.email,
+				contactName: inputs.contact_name,
+			},
+			category: inputs.category,
+			subcategory: inputs.subcategory,
+		};
+		console.log('submit_site -> postData:', postData);
+		const req = {
+			url: `${API_HOST}/organizations/${inputs.org_id}/sites`,
+			data: JSON.stringify(postData),
+			type: 'POST',
+			success: responseData => {
+				console.log('submit_site -> responseData:', responseData);
+				const emailDomain = inputs.email.split('@')[1];
+				const link = `https://${emailDomain}`;
+
+				$(inputs.button_id).removeClass('btn-loading');
+				$(inputs.modal_id).find('.alert-warning').addClass('d-none');
+				$(inputs.modal_id).find('.alert-success a').attr('href', link).attr("target", "_blank");
+				$(inputs.modal_id).find('.alert-success').removeClass('d-none');
+				$(inputs.button_id).replaceWith(`<a href="${link}", target="_blank", class="btn btn-secondary btn-block">Check your email</a>`);
+			},
+			error: e => {
+				console.log('Error from "submit_site":', e);
+				console.log('Arguments from "submit_site":', inputs);
+				$(inputs.button_id).removeClass('btn-loading');
+				$(inputs.modal_id).find('.alert-warning').removeClass('d-none');
+			},
+		};
+		jQuery.ajax(Object.assign(defaultRequestOpts, req));
+	},
+	scan: (inputs, callback) => {
+		console.log('scan -> inputs:', inputs);
+		const scannedId = localStorage.getItem('dvid');
```

Comment:
Surprisingly this is the one instance typescript is helpful 😅
should be:
`const scannedID = inputs.sdvid`
-------------------------------------------
Some Discussion:
`const scannedID = localStorage.getItem('dvid'),` means the user is just checking in with themselves

Wonder if there is some ES6 notation where we can declare the fields in for the input object. In typescript it was as such:
`inputs: { sdvid?: string; ip?: string; fingerprint?: string } | undefined,`

Since the inputs object does change as it passes down the callback chain:
input: {sdvid} 
fetch IP -> input: { ip?: string; sdvid?: string }
fingerprint -> input: {ip?: string; sdvid?: string; fingerprint?: string},
create ->  input: {ip?: string; sdvid?: string; fingerprint?: string},

sdvid is passed down the callback chain through the input object on controller.ts line 75.
1. sdvid is horrible naming, thats on me -> should probably be named scannedID instead
2. The callback chain is a little annoying but I avoided using Promises out of browser coverage fears

Reply:
You're right. We should definitely clean up/re-name function arguments. If we're going to conver to React we could hold off until then, just make sure our new functions are more explicit.

I'll look into modifying the function to only take specific arguments. Or, we can do de-structuring in the function itself like

`scan({ sdvid, ip, fingerprint}, callback)` That would make it more explicit, but wouldn't have any of the type-enforcement like with TypeScript.

Reply:
Awesome, seems like for now de-structuring would better define what inputs are being expected. Thank you!